ddtrace/debugging/_probe/model.py

```diff
@@ -227,73 +162,30 @@ def eval(self, _locals):
 
 @attr.s
 class ExpressionTemplateSegment(TemplateSegment):
-    expr = attr.ib(type=DslExpression, default=None)  # type: DslExpression
+    expr = attr.ib(type=DDExpression, default=None)  # type: DDExpression
 
     def eval(self, _locals):
         # type: (Dict[str,Any]) -> Any
         return self.expr.eval(_locals)
 
 
 @attr.s
-class SnapshotProbeDetails(six.with_metaclass(abc.ABCMeta)):
-    capture = attr.ib(type=CaptureLimits, eq=False)  # type: CaptureLimits
-
-
-def snapshot_probe_defaults(f):
-    def _wrapper(*args, **kwargs):
-        kwargs.setdefault("capture", CaptureLimits())
-        return f(*args, **kwargs)
-
-    return _wrapper
+class LogProbeMixin(object):
+    template = attr.ib(type=str)
+    segments = attr.ib(type=List[TemplateSegment])
+    capture_snapshot = attr.ib(type=bool)
+    limits = attr.ib(type=CaptureLimits, eq=False)  # type: CaptureLimits
 
 
 @attr.s
-class SnapshotLineProbe(Probe, ProbeConditionDetails, LineLocationDetails, SnapshotProbeDetails):
-    @classmethod
-    @create_probe_defaults
-    @probe_conditional_defaults
-    @snapshot_probe_defaults
-    def create(cls, **kwargs):
-        return SnapshotLineProbe(**kwargs)
+class LogLineProbe(Probe, LineLocationMixin, LogProbeMixin, ProbeConditionMixin):
+    pass
 
 
 @attr.s
-class SnapshotFunctionProbe(Probe, ProbeConditionDetails, FunctionLocationDetails, SnapshotProbeDetails):
-    @classmethod
-    @create_probe_defaults
-    @probe_conditional_defaults
-    @function_location_defaults
-    @snapshot_probe_defaults
-    def create(cls, **kwargs):
-        return SnapshotFunctionProbe(**kwargs)
+class LogFunctionProbe(Probe, FunctionLocationMixin, LogProbeMixin, ProbeConditionMixin):
+    pass
 
 
-@attr.s
-class LogProbeDetails(six.with_metaclass(abc.ABCMeta)):
-    template = attr.ib(type=Optional[str])
-    segments = attr.ib(type=Optional[List[TemplateSegment]])
-
-
-@attr.s
-class LogLineProbe(Probe, ProbeConditionDetails, LineLocationDetails, LogProbeDetails, SnapshotProbeDetails):
-    @classmethod
-    @create_probe_defaults
-    @probe_conditional_defaults
-    @snapshot_probe_defaults
-    def create(cls, **kwargs):
-        return LogLineProbe(**kwargs)
-
-
-@attr.s
-class LogFunctionProbe(Probe, ProbeConditionDetails, FunctionLocationDetails, LogProbeDetails, SnapshotProbeDetails):
-    @classmethod
-    @create_probe_defaults
-    @probe_conditional_defaults
-    @function_location_defaults
-    @snapshot_probe_defaults
-    def create(cls, **kwargs):
-        return LogFunctionProbe(**kwargs)
-
-
-LineProbes = Union[SnapshotLineProbe, LogLineProbe, MetricLineProbe]
-FunctionProbes = Union[SnapshotFunctionProbe, LogFunctionProbe, MetricFunctionProbe]
+LineProbes = Union[LogLineProbe, MetricLineProbe]
+FunctionProbes = Union[LogFunctionProbe, MetricFunctionProbe]
```

Comment:
I think we might want to use the singular for defining type aliases
```suggestion
LineProbe = Union[LogLineProbe, MetricLineProbe]
FunctionProbe = Union[LogFunctionProbe, MetricFunctionProbe]
```

Reply:
renamed